### PR TITLE
feat: Propagator optionally inherits from BasePropagator

### PR DESCRIPTION
**Pull request by paulgessinger**
This enables components which only need bound parameter propagation (like the vertexing in #2842) to use an interface rather than the concrete type.

Propagator+Stepper combinations need to be flagged as supporting bound parameter propagation, e.g. the default EigenStepper supports this, while the EigenStepper with the dense extension does not.

I did not find a good way to determine this automatically at compile-time, so I added the flagging mechanism.

The base propagator interface looks like this:

```cpp
class BasePropagator {
 public:
  using Options = PropagatorOptions<>;

  virtual Result<BoundTrackParameters> propagate(
      const BoundTrackParameters& start, const Surface& target,
      const Options& options) const = 0;
};

```

### CODE REVIEW

File: Core/include/Acts/Propagator/Propagator.hpp [hunk 1]
```diff
@@ -178,6 +180,52 @@ struct PropagatorOptions : public PropagatorPlainOptions {
   std::reference_wrapper<const MagneticFieldContext> magFieldContext;
 };
 
+/// Common simplified base interface for propagators.
+///
+/// This class only supports propagation from start bound parameters to a target
+/// surface and returns only the end bound parameters.
+/// Navigation is performed if the underlying propagator is configured with an
+/// appropriate navigator. No custom actors or aborters are supported.
+class BasePropagator {
```
Comment (andiwand) [inline Core/include/Acts/Propagator/Propagator.hpp:189]: should be define a virtual destructor?

File: Core/include/Acts/Propagator/EigenStepper.hpp [hunk 2]
```diff
@@ -420,6 +422,11 @@ class EigenStepper {
   /// Overstep limit
   double m_overstepLimit;
 };
+
+template <typename navigator_t>
+struct supportsBoundParameters<EigenStepper<>, navigator_t>
```
Comment (andiwand) [inline Core/include/Acts/Propagator/EigenStepper.hpp:427]: just noticed the starting lower-case. is this what we do with traits?

File: Core/include/Acts/Propagator/Propagator.hpp [hunk 3]
```diff
@@ -178,6 +180,52 @@ struct PropagatorOptions : public PropagatorPlainOptions {
...
+      const Options& options) const = 0;
+};
+
+namespace detail {
+class PropagatorStub {};
+
+template <typename derived_t>
+class BasePropagatorHelper : public BasePropagator {
+ public:
+  Result<BoundTrackParameters> propagate(
```
Comment (andiwand) [inline Core/include/Acts/Propagator/Propagator.hpp:210]: could potentially be moved to ipp

File: Core/include/Acts/Propagator/Propagator.hpp [hunk 4]
```diff
@@ -178,6 +180,52 @@ struct PropagatorOptions : public PropagatorPlainOptions {
...
+/// Common simplified base interface for propagators.
+///
+/// This class only supports propagation from start bound parameters to a target
+/// surface and returns only the end bound parameters.
+/// Navigation is performed if the underlying propagator is configured with an
+/// appropriate navigator. No custom actors or aborters are supported.
+class BasePropagator {
+ public:
+  /// Base propagator options
+  using Options = PropagatorOptions<>;
```
Comment (andiwand) [inline Core/include/Acts/Propagator/Propagator.hpp:192]: curvilinear overload does not make sense here because we locked down the options right?

**Review by andiwand — COMMENTED** [review]
cool!

File: Core/include/Acts/Propagator/Propagator.hpp (hunk 4 above)
Comment (paulgessinger) [reply Core/include/Acts/Propagator/Propagator.hpp:192]: You mean curvilinear start params?

File: Core/include/Acts/Propagator/EigenStepper.hpp (hunk 2 above)
Comment (paulgessinger) [reply Core/include/Acts/Propagator/EigenStepper.hpp:427]: Maybe not. I'll change it.

File: Core/include/Acts/Propagator/Propagator.hpp (hunk 4 above)
Comment (andiwand) [reply Core/include/Acts/Propagator/Propagator.hpp:192]: like here with final curvilinear params

https://github.com/acts-project/acts/blob/cbd38cd023e71a189b9845c4932c104f12cb7b28/Core/include/Acts/Propagator/Propagator.hpp#L374-L380

Comment (paulgessinger) [reply Core/include/Acts/Propagator/Propagator.hpp:192]: Added a line to the test which executes this. So I think this works.

**Review by andiwand — DISMISSED** [review]

**Comment by paulgessinger** [comment]
Ok there's a build failure I have to investigate.

**Comment by paulgessinger** [comment]
@andiwand could you go one more time?

**Review by andiwand — APPROVED** [review]
:+1:

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [a8c79de1966db214563774b8194aad54da09c7c3]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395967)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395966)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395965)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395964)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395963)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395962)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395961)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395960)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395959)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395958)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395957)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395956)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395955)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395954)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395953)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395952)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395951)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395950)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395949)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395948)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395947)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395946)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35395945)||

**Pull request merged**